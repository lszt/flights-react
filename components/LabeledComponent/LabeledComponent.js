import React, { PropTypes, Component } from 'react';
import './LabeledComponent.scss';

class LabeledComponent extends Component {

  static propTypes = {
    label: PropTypes.string,
    className: PropTypes.string,
    component: PropTypes.element,
  };

  render() {
    const className = 'LabeledComponent ' + this.props.className;
    return (
      <div className={className}>
        <label>{this.props.label}</label>
        <div className="component-wrap">{this.props.component}</div>
      </div>
    );
  }
}

export default LabeledComponent;
