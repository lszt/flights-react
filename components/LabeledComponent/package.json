{
  "name": "LabeledComponent",
  "version": "0.0.0",
  "private": true,
  "main": "./LabeledComponent.js"
}
