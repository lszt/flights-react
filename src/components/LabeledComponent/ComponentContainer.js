import React from 'react';
import styled from 'styled-components';

const ComponentContainer = styled.div`
  padding: 0.2em;
  
  & > * {
    font-size: 1.5em;
    width: 100%;
    resize: none;
  }
`;

export default ComponentContainer;
