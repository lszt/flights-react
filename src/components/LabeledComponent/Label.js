import React from 'react';
import styled from 'styled-components';

const Label = styled.label`
  margin-bottom: 1.5em;
  display: block;
`;

export default Label;
