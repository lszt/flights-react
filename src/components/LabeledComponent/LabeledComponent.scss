.LabeledComponent {

  .component-wrap {
    padding: 0.2em;

    > * {
      font-size: 1.5em;
      border: 0;
      width: 100%;
      resize: none;
    }
  }

  .validation-msg {
    color: #ed351c;
    overflow: hidden;

    .icon {
      float: left;
    }

    .text {
      font-style: italic;
      line-height: 1.3em;
      margin-left: 2em;
    }
  }
}
