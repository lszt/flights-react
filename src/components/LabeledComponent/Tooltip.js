import React from 'react';
import styled from 'styled-components';

const Tooltip = styled.div`
  position: absolute;
  bottom: 55px;
  margin-left: auto;
  margin-right: auto;
  left: 0;
  right: 0;
  background-color: #4d4d4d;
  color: #fff;
  border-radius: 5px;
  padding: 9px;
  line-height: 1.4em;
  font-size: 0.9em;
  
  &:after {
    border-top: 8px solid #4d4d4d;
    border-left: 10px solid transparent;
    border-right: 10px solid transparent;
    bottom: -8px;
    left: 50%;
    margin-left: -10px;
    content: "";
    width: 0;
    height: 0;
    position: absolute;
  }
`;

export default Tooltip;
