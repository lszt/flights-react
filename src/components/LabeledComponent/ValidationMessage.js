import React from 'react';
import styled from 'styled-components';
import MaterialIcon from '../MaterialIcon';

const Wrapper = styled.div`
  color: ${props => props.theme.colors.danger};
  overflow: hidden;
`;

const Icon = styled(MaterialIcon)`
  float: left;
`;

const Text = styled.div`
  font-style: italic;
  line-height: 1.3em;
  margin-left: 2em;
`;

class ValidationMessage extends React.PureComponent {

  render() {
    return (
      <Wrapper>
        <Icon icon="error"/>
        <Text>{this.props.error}</Text>
      </Wrapper>
    );
  }
}

ValidationMessage.propTypes = {
  error: React.PropTypes.string.isRequired,
};

export default ValidationMessage;
