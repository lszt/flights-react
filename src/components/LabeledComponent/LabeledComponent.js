import React, { PropTypes, Component } from 'react';
import Label from './Label';
import ComponentContainer from './ComponentContainer';
import Tooltip from './Tooltip';
import ValidationMessage from './ValidationMessage';
import Wrapper from './Wrapper';

class LabeledComponent extends Component {

  constructor(props) {
    super(props);
    this.state = {
      tooltipVisible: false,
    };
    this.handleFocus = this.handleFocus.bind(this);
    this.handleBlur = this.handleBlur.bind(this);
  }

  render() {
    return (
      <Wrapper className={this.props.className} onFocus={this.handleFocus} onBlur={this.handleBlur}>
        <Label>{this.props.label}</Label>
        {this.props.validationError && <ValidationMessage error={this.props.validationError}/>}
        <ComponentContainer>{this.props.component}</ComponentContainer>
        {this.state.tooltipVisible && this.props.tooltip && <Tooltip>{this.props.tooltip}</Tooltip>}
      </Wrapper>
    );
  }

  handleFocus() {
    this.setState({
      tooltipVisible: true,
    });
  }

  handleBlur() {
    this.setState({
      tooltipVisible: false,
    });
  }
}

LabeledComponent.propTypes = {
  label: PropTypes.string.isRequired,
  component: PropTypes.element.isRequired,
  className: PropTypes.string,
  validationError: PropTypes.string,
  tooltip: PropTypes.string,
};

export default LabeledComponent;
