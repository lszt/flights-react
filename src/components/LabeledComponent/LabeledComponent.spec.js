import React from 'react';
import {shallow} from 'enzyme';
import LabeledComponent from './LabeledComponent';

describe('components', () => {
  describe('LabeledComponent', () => {
    it('is built correctly', () => {
      const component = shallow(<LabeledComponent label="My label" component={<input type="text" id="my-input"/>}/>);
      expect(component.debug()).toMatchSnapshot();
    });
  });
});
