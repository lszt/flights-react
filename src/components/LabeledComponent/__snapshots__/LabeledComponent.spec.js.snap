// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components LabeledComponent is built correctly 1`] = `
"<styled.div className={[undefined]} onFocus={[Function: bound handleFocus]} onBlur={[Function: bound handleBlur]}>
  <styled.label>
    My label
  </styled.label>
  <styled.div>
    <input type=\\"text\\" id=\\"my-input\\" />
  </styled.div>
</styled.div>"
`;
